src/Context.ts

```diff
@@ -87,31 +87,35 @@ class Context implements IContext {
   }
 }
 
-function getLogger(invocationId: string, functionName: string, log: ILogCallback): ILogger{
+function getLogger(invocationId: string, functionName: string, log: LogCallback): Logger{
     return Object.assign(
-      <ILog>(...args: any[]) => log(LogLevel.Information, ...args),
+      <Log>(...args: any[]) => log(LogLevel.Information, ...args),
       {
-        error: <ILog>(...args: any[]) => log(LogLevel.Error, ...args),
-        warn: <ILog>(...args: any[]) => log(LogLevel.Warning, ...args),
-        info: <ILog>(...args: any[]) => log(LogLevel.Information, ...args),
-        verbose: <ILog>(...args: any[]) => log(LogLevel.Trace, ...args)
+        error: <Log>(...args: any[]) => log(LogLevel.Error, ...args),
+        warn: <Log>(...args: any[]) => log(LogLevel.Warning, ...args),
+        info: <Log>(...args: any[]) => log(LogLevel.Information, ...args),
+        verbose: <Log>(...args: any[]) => log(LogLevel.Trace, ...args)
       }
     );
 }
 
-export interface IInvocationResult {
+export interface InvocationResult {
   return: any;
-  bindings: IDict<any>;
+  bindings: Dict<any>;
 }
 
-export interface ILogCallback {
+export interface DoneCallback {
+  (err?: any, result?: any): void;
+}
+
+export interface LogCallback {
   (level: LogLevel, ...args: any[]): void;
 }
 
-export interface IResultCallback {
-  (err?: any, result?: IInvocationResult): void;
+export interface ResultCallback {
+  (err?: any, result?: InvocationResult): void;
 }
 
-export interface IDict<T> {
+export interface Dict<T> {

```

Comment:
See comment above for discussion on the Dict type. This is an overly broad type that will invite some potential issues in the future if you want to change it.

Reply:
Same as previous comment, will push Dict type changes to future change since it isn't public facing